confluence-syntax-xhtml/src/test/resources/confluence+xhtml10/specific/macro/macro4.test

```diff
@@ -0,0 +1,12 @@
+.#-----------------------------------------------------
+.input|confluence+xhtml/1.0
+.# If a macro parameter content in Confluence is a user reference, the parameter value of the generated wiki macro call is empty
+.#-----------------------------------------------------
+<ac:structured-macro ac:name="profile" ac:schema-version="1"><ac:parameter ac:name="user"><ri:user ri:userkey="0123456789abcdefedcba98765432100" /></ac:parameter></ac:structured-macro><ac:structured-macro ac:name="profile" ac:schema-version="1"><ac:parameter ac:name="user"><ri:user ri:username="user1" /></ac:parameter></ac:structured-macro>
+.#-----------------------------------------------------
+.expect|event/1.0
+.#-----------------------------------------------------
+beginDocument
+onMacroStandalone [profile] [att--user=0123456789abcdefedcba98765432100|user=]
```

Comment:
I don't understand this result: the name of the macro parameter being "user" that what I would expect to find in the events.

Reply:
It is the same as I did for attachments. I was unsure if there could be a macro where there is content besides the user reference/attachment, which would (as far as I have understood) be put in the original parameter the reference/attachment was contained in. In that case simply assigning the attachment filename/user reference to the original parameter would overwrite the other content (or get overwritten by it), so I decided to take the safe and simple route and assign everything with special handling to a separate variable.

There are probably better ways to handle this. Maybe appending to the original parameter or adding the filename/reference to the context (I tried the latter but did not get it to work). Actually there is a good chance that there never are parameters with extra content, but I can't be sure.

Reply:
I see, but the logic behind the "att--user" format is not very clear to me.

Reply:
Ok, I understand. Its actually a remnant of the fix for attachment filenames which I called 'att--filename'. For the users I did not know what I should prefix them with and just left the 'att' in there. Maybe something like 'conv' (converted), 'tag' or 'ext' (extra,external) would be clearer? I chose double dashes instead of a single dash because I wanted to make sure that I would not overwrite any potentially existing parameter. I expected that a single dash might be used in some parameter names, but that there is a high probability nobody would choose double dashes for a parameter name.

Reply:
What about something like "." as prefix (so `.user`) to really make it look special and a bit internal ?

Reply:
Sounds good. The dot character will not cause any problems in parameter names? Just double checking before making the change.

Reply:
Won't be a problem for a `MacroConverter` at least. The "." prefix makes it a little bit less obvious to define a parameter for a XWiki macro implemented in Java, but we might see this as kind of feature (discourage relying on it).

Reply:
Ok, pushed.